core/templates/domain/admin/admin-backend-api.service.ts

```diff
@@ -140,6 +147,59 @@ export class AdminBackendApiService {
       });
     });
   }
+
+  private _postAdminActionAsync(action:string, payload:Object): Promise<void> {
+    return this.http.post<void>(
+      AdminPageConstants.ADMIN_HANDLER_URL, { action, ...payload }).toPromise();
+  }
+
+  async startNewJobAsync(jobType: string): Promise<void> {
+    let action = 'start_new_job';
+    let payload = {
+      job_type: jobType
+    };
+    return this._postAdminActionAsync(action, payload);
+  }
+
+  async cancelJobAsync(jobId: string, jobType: string): Promise<void> {
+    let action = 'cancel_job';
+    let payload = {
+      job_id: jobId,
+      job_type: jobType
+    };
+    return this._postAdminActionAsync(action, payload);
+  }
+
+  async startComputationAsync(computationType: string): Promise<void> {
+    let action = 'start_computation';
+    let payload = {
+      computation_type: computationType
+    };
+    return this._postAdminActionAsync(action, payload);
+  }
+
+  async stopComputationAsync(computationType: string): Promise<void> {
+    let action = 'stop_computation';
+    let payload = {
+      computation_type: computationType
+    };
+    return this._postAdminActionAsync(action, payload);
+  }
+
+  async showJobOutputAsync(jobId: string): Promise<string> {
+    let adminJobOutputUrl = this.urlInterpolationService.interpolateUrl(
+      AdminPageConstants.ADMIN_JOB_OUTPUT_URL_TEMPLATE, {
+        jobId: jobId
+      });
+    return new Promise((resolve, reject) => {
+      this.http.get<JobOutput>(
+        adminJobOutputUrl).toPromise().then(response => {
+        resolve(response.output);
+      }, errorResponse =>{
+        reject(errorResponse);
+      });
+    });
```

Comment:
```suggestion
    return this.http.get<JobOutput>(adminJobOutputUrl).toPromise()
      .then(response => {
        return response.output;
      }, errorResponse => {
        return errorResponse;
      }
    );
```

Will this^ work? (not sure!)

Reply:
done

Reply:
@DubeySandeep although Fe is working fine with this change in promises,
FE tests are failing(function are not even getting called )
they are expecting resolve, reject for the return promise
thoughts?

Reply:
@Radesh-kumar, I'm not sure what's going on here, is it anything released to angular to angularJS issue? 

@srijanreddy98, do you have any thoughts?